### ISSUE TITLE
RFC: Remix Support

### ISSUE THREAD

**Issue by RichiCoder1** [StellateHQ/fuse.js#88]
## Summary

Official request for [Remix](https://remix.run) support.

## Proposed Solution

In theory Next.js API Routes are analogous to [Resource Routes](https://remix.run/docs/en/main/guides/resource-routes) and Service Side Props to [Loaders](https://remix.run/docs/en/main/route/loader) so a lot of the concepts already present should translate.

**Comment by JoviDeCroock** [comment]
Hey,

That's a great suggestion 🙌 , I do think with Remix moving to `vite` this could become pretty convenient as we have already explored `vite` in #79. We can use `import.meta.glob` to add the schema-types declared in the `types/` directory and create a way to initialise the `resource route`.

The exploration after that would be how it look on the client-side, I personally don't have too much experience with Remix so I'm going to have to do some reading at first before I can participate in discussing how this API surface looks.

For generating the client-side types I reckon that we can hook into the `vite` plugin lifecycle to boot the code-generator.

**Comment by RichiCoder1** [comment]
Yup, Remix moving to `vite` is a massive enabler for automatic generation (and also other non-remix projects). I've been using this [vite graphql codegen plugin](https://github.com/danielwaltz/vite-plugin-graphql-codegen) with Pothos in a side project and it works pretty well. So would love to drop all that glue for an opinionated solution! As a bonus, Remix [is considering RSC](https://remix.run/blog/react-server-components) so should align on that goal eventually too.
